plugins/policy/processor/processor.go

```diff
@@ -416,22 +416,23 @@ func (pp *PolicyProcessor) getPoliciesAssignedToPod(pod *podmodel.Pod) (policies
 		} else {
 			for _, ingressRules := range dataPolicy.IngressRule {
 				for _, ingressRule := range ingressRules.From {
-					matchLabels := []*policymodel.Policy_Label{}
-					matchExpressions := []*policymodel.Policy_LabelSelector_LabelExpression{}
+					matchPodSelectorLabels := []*policymodel.Policy_Label{}
+					matchPodSelectorExpressions := []*policymodel.Policy_LabelSelector_LabelExpression{}
+					matchNsSelectorLabels := []*policymodel.Policy_Label{}
+					matchNsSelectorExpressions := []*policymodel.Policy_LabelSelector_LabelExpression{}
 
 					if ingressRule.Pods != nil {
-						matchLabels = ingressRule.Pods.MatchLabel
-						matchExpressions = ingressRule.Pods.MatchExpression
+						matchPodSelectorLabels = ingressRule.Pods.MatchLabel
+						matchPodSelectorExpressions = ingressRule.Pods.MatchExpression
 					}
-					isMatchPodSelector := pp.calculateLabelSelectorMatches(pod, matchLabels, matchExpressions, dataPolicy.Namespace)
 
 					if ingressRule.Namespaces != nil {
-						matchLabels = ingressRule.Namespaces.MatchLabel
-						matchExpressions = ingressRule.Namespaces.MatchExpression
+						matchNsSelectorLabels = ingressRule.Namespaces.MatchLabel
+						matchNsSelectorExpressions = ingressRule.Namespaces.MatchExpression
 					}
 
-					isMatchNamespaceSelector := pp.isNamespaceMatchLabel(pod, matchLabels)
-					if !isMatchPodSelector && !isMatchNamespaceSelector {
+					if !pp.isPodLabelSelectorMatch(pod, matchPodSelectorLabels, matchPodSelectorExpressions, dataPolicy.Namespace) &&
```

Comment:
This condition always returns false, because one of the function is always called with empty match* arrays => returns true.
See policy.proto, exactly one item in struct Peer is non-nil, the rest are nils.